test/AutoRest.TestServer.Tests/Common/Utilities/NamedTypeSymbolExtensionsTests.cs

```diff
@@ -159,12 +159,12 @@ public void IsSameType_PrimitiveTypes()
         public void IsSameType_ModelTypes()
         {
             // Different namespace
-            var input = new InputModelType("MetadataModel", "", null, null, null, InputModelTypeUsage.RoundTrip, null, null, Array.Empty<InputModelType>(), null, null, null, null);
+            var input = new InputModelType("MetadataModel", "", null, null, null, InputModelTypeUsage.Input | InputModelTypeUsage.Output, null, null, Array.Empty<InputModelType>(), null, null, null, null);
```

Comment:
I do not like it because the input type csharp types should be reflecting the structure as the models defined in TCGC exactly.
We should not include extra public members to them since they are public and cannot be changed once added.
If you make the statistics, majority of the usage is in test cases.
Test cases could do something, but we should not add it into the InputModelTypeUsage API.

And since I have done all the mitigations of those usages, I do not see it is needed we have to introduce the convenience API for roundtrip in this PR right now.